.gitignore

```diff
@@ -70,9 +70,11 @@ cpp/thirdparty/googletest/
 *.iws
 
 ## Datasets
-datasets/*
+dataset/
+datasets/
```

Comment:
Do we really need both of these paths? Why?

Reply:
Good point, there's an opportunity to simplify here.

I added `dataset/` because I saw files created at `python/cugraph-pyg/cugraph_pyg/dataset/` when running the `cugraph-pyg` examples locally.

Looks like that comes from here: https://github.com/snap-stanford/ogb/blob/f631af76359c9687b2fe60905557bbb241916258/ogb/nodeproppred/dataset.py#L13

I just pushed changes writing all those datasets to `datasets/` instead, so we don't need to have both rules in gitignore.